examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -65,26 +84,29 @@ NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
     goto out;
   }
 
+  struct bcc_symbol resolved_symbol;
+  if (NativeStackTrace::insert_dso_name) {
+    procSymbols = get_proc_symbols(pid);
+  }
+
   do {
     unw_word_t offset;
-    char sym[256];
-    char   *realname;
-    int     status;
-
-    // TODO: This function is very heavy. We should try to do some caching here, maybe in the
-    //       underlying UPT function.
-    res = unw_get_proc_name(&cursor, sym, sizeof(sym), &offset);
-    if (res == 0) {
-      realname = abi::__cxa_demangle(sym, NULL, NULL, &status);
-      if (status == 0) {
-        this->symbols.push_back(std::string(realname));
-      } else {
-        this->symbols.push_back(std::string(sym));
-      }
-      free(realname);
+    unw_word_t ip;
+    unw_get_reg(&cursor, UNW_REG_IP, &ip);
+    if (procSymbols && procSymbols->resolve_addr(ip, &resolved_symbol, true)) {
+        snprintf(buf, buf_size, "%s (%s)", resolved_symbol.demangle_name, resolved_symbol.module);
+        this->symbols.push_back(std::string(buf));
+    } else if (!unw_get_proc_name(&cursor, sym, sizeof(sym), &offset)) {
```

Comment:
```suggestion
    } else if (!unw_get_proc_name(&cursor, buf, sizeof(buf), &offset)) {
```

for what it's worth, we already have a larger buffer, let's use it and avoid getting truncated names :)

Reply:
You can remove `sym` altogether I guess

Reply:
👍